### ISSUE TITLE
De-duplicate event configuration?

### ISSUE THREAD

**Issue by KevinBongart** [aasm/aasm#295]
Hi, I'm seeing a lot of duplicated code when configuring AASM. For instance, based on the README:

```ruby
class Job
  include AASM

  aasm do
    state :sleeping, :initial => true
    state :running
    state :cleaning

    event :run do
      transitions :from => :sleeping, :to => :running
      error { |e| raise SomeSpecificError.new("#{e.message} (id: #{self.id})") }
    end

    event :clean do
      transitions :from => :running, :to => :cleaning
      error { |e| raise SomeSpecificError.new("#{e.message} (id: #{self.id})") }
    end

    event :sleep do
      transitions :from => [:running, :cleaning], :to => :sleeping
      error { |e| raise SomeSpecificError.new("#{e.message} (id: #{self.id})") }
    end
  end
end
```

Is there a way to avoid duplication with something like:

```ruby
class Job
  include AASM

  aasm do
    state :sleeping, :initial => true
    state :running
    state :cleaning

    all_events do
      error { |e| raise SomeSpecificError.new("#{e.message} (id: #{self.id})") }
    end

    event :run do
      transitions :from => :sleeping, :to => :running
    end

    event :clean do
      transitions :from => :running, :to => :cleaning
    end

    event :sleep do
      transitions :from => [:running, :cleaning], :to => :sleeping
    end
  end
end
```

or something similar to FactoryGirl's DSL:

```ruby
class Job
  include AASM

  aasm do
    state :sleeping, :initial => true
    state :running
    state :cleaning

    base_event do
      error { |e| raise SomeSpecificError.new("#{e.message} (id: #{self.id})") }
    end

    event :run, parent: :base_event do
      transitions :from => :sleeping, :to => :running
    end

    event :clean, parent: :base_event do
      transitions :from => :running, :to => :cleaning
    end

    event :sleep, parent: :base_event do
      transitions :from => [:running, :cleaning], :to => :sleeping
    end
  end
end
```

What options do I have to de-duplicate event configuration?

Thanks

**Comment by HoyaBoya** [comment]
@KevinBongart my hope is that my PR in #282 could alleviate this condundrum.

**Comment by KevinBongart** [comment]
@HoyaBoya This is exactly what I was looking for! Let's get this merged :)

**Comment by alto** [comment]
@KevinBongart I merged #282 and will release a new version soon.

**Comment by KevinBongart** [comment]
Awesome, thanks!

**Comment by alto** [comment]
Released version `4.7.0` including the addition.
